packages/components/segmented/src/segmented.vue

```diff
@@ -74,10 +74,21 @@ const state = reactive({
   focusVisible: false,
 })
 
-const handleChange = (item: Option) => {
+const handleChange = (item: Option, evt: Event) => {
   const value = getValue(item)
   emit(UPDATE_MODEL_EVENT, value)
   emit(CHANGE_EVENT, value)
+
+  if (!getSelected(item)) {
+    evt.target && ((evt.target as HTMLInputElement).checked = false)
+    if (!segmentedRef.value) return
+    const selectedItemInput = segmentedRef.value.querySelector(
+      '.is-selected input'
+    ) as HTMLInputElement
+    if (selectedItemInput) {
+      selectedItemInput.checked = true
+    }
+  }
```

Comment:
It seems to follow the radio component on change.
https://github.com/element-plus/element-plus/blob/dev/packages/components/radio/src/use-radio.ts#L37

reason: By default clicking on a radio element will set checked=true without any restriction. Am i wrong ? @ykcory

I think it can be simplified:
```suggestion
  ;(evt.target as HTMLInputElement).checked = value === props.modelValue
```